### ISSUE TITLE
Core React component AI enhancements (phase 2)

### ISSUE THREAD

**Issue by mattrosno** [carbon-design-system/carbon#15460]
This issue tracks the effort to add AI enhancements (slug and gradient) to the next wave of core components for phase 2. On a per-component basis, once the enhancement has been made to a React component (via the styles package), that component can then be enhanced in the web component.

**Acceptance criteria**

- Enhancements designated with experimental status

**Out of scope**

- T~his effort is not the "full AI" theme that has layer styling like opacity and blur~ 
- Usage guidance, website documentation, Figma libraries, and other items required for stable status (graduation from experimental) will happen after phase 2

```[tasklist]
### Design
- [ ] https://github.com/carbon-design-system/carbon/issues/15620
- [ ] https://github.com/carbon-design-system/carbon/issues/15463
- [ ] https://github.com/carbon-design-system/carbon/issues/15619
- [ ] https://github.com/carbon-design-system/carbon/issues/15615
- [ ] https://github.com/carbon-design-system/carbon/issues/15301
- [ ] https://github.com/carbon-design-system/carbon/issues/15461
- [ ] https://github.com/carbon-design-system/carbon/issues/15464
- [ ] https://github.com/carbon-design-system/carbon/issues/15465
- [ ] https://github.com/carbon-design-system/carbon/issues/15466
- [ ] https://github.com/carbon-design-system/carbon/issues/15467
- [ ] https://github.com/carbon-design-system/carbon/issues/15468
```

```[tasklist]
### Development
- [ ] https://github.com/carbon-design-system/carbon/issues/15617
- [ ] https://github.com/carbon-design-system/carbon/issues/15469
- [ ] https://github.com/carbon-design-system/carbon/issues/15470
- [ ] https://github.com/carbon-design-system/carbon/issues/15471
- [ ] https://github.com/carbon-design-system/carbon/issues/15472
- [ ] https://github.com/carbon-design-system/carbon/issues/15473
- [ ] https://github.com/carbon-design-system/carbon/issues/15474
- [ ] https://github.com/carbon-design-system/carbon/issues/15475
- [ ] https://github.com/carbon-design-system/carbon/issues/15476
```

**Comment by mattrosno** [comment]
@aagonzales I'll make sure we have a separate issue for the React modal updates.

I've closed everything not done as not planned for phase 2. Is there anything in this epic that needs done by 2/23? If no, let's close this epic as complete.
